main/src/cgeo/geocaching/utils/RxOkHttpUtils.java

```diff
@@ -35,8 +36,16 @@ public void subscribe(final SingleEmitter<Response> singleEmitter) throws Except
                 singleEmitter.setDisposable(Disposables.fromRunnable(new Runnable() {
                     @Override
                     public void run() {
-                        if (!completed.get()) {
-                            call.cancel();
+                        if (!completed.getAndSet(true)) {
+                            // Cancel request on a non-UI thread as operations such as close()
```

Comment:
I suggest to replace this explicit scheduling by a call to a new method, and to mark that method @UiThread (from androidannotations). That is more easy to read and understand.